clash-lib/src/Clash/Netlist/BlackBox/Util.hs

```diff
@@ -476,6 +492,12 @@ renderElem b (IF c t f) = do
                       Literal {}   -> 1
                       BlackBoxE {} -> 1
                       _            -> 0
+      (IsScalar n) -> let (_,ty,_) = bbInputs b !! n
+                      in case ty of
+                        -- See NOTE [scalar netlist types]
```

Comment:
The type of this function is:
```
renderElem
  :: HasCallStack
  => Backend backend
  => BlackBoxContext
  -> Element
  -> State backend (Int -> Text)
```
i.e. we get access to the `Backend backend => backend`. This means we get access to https://github.com/clash-lang/clash-compiler/blob/ee31c20e73516cffa396eb4720083acfbe10fd42/clash-lib/src/Clash/Backend.hs#L94 which allows us to differentiate between the HDL we're rendering for. This in turns means we can decide what is a scalar or not depending on the hdl we're generating, obviating the need for the note, and just implementing it properly.

Reply:
I have implemented it. I'm no Verilog/VHDL star but from the specs I can read online in (System)Verilog only `Bit` and `Bool` are scalar. in VHDL in addition to those `integer`, `real`, enumeration and physical types are scalar. But we never generate `real` or physical types.